### DISCUSSION TITLE
Objective function

### DISCUSSION THREAD

**Discussion by ndkazu** [mdolab/dafoam#531] [General]
What parameter allows you to determine if you want to Minimize or Maximize an objective function?
is it the `scaler `in: `self.add_objective("cruise.aero_post.MFR", scaler=1.0)` ? if yes what are the value corresponding to each configuration?

**Comment by friedenhe** [comment]
Yes. scaler=1.0 means minimizing MFR, and scaler=-1.0 means maximizing MFR.

**Reply by 15175829310** [reply]
I am using the field inversion script, and I found that the self.add_objective("cruise.aero_post.FI", scaler=1.0) is in runscript.py, but it is not in runscript_v2.py, so how do you determine maximization and minimization in v2.py?

**Comment by friedenhe** [comment]
it is called `addObj`

**Reply by 15175829310** [reply]
Is that the one？Do I need to add scaler=1.0 to it? And what does scale mean in this case?
![Snipaste_2024-03-12_10-26-51](https://github.com/mdolab/dafoam/assets/102606153/cd6bd3ec-b5b9-4e4c-97e6-81e3ea9af3ed)

**Comment by friedenhe** [comment]
Yes. `scale` is for v2 script. In v3 scripts, it is called `scaler`. They are the same.
